hypothesis-python/src/hypothesis/internal/conjecture/shrinker.py

```diff
@@ -1144,7 +1174,34 @@ def adaptive_example_deletion(self):
         If we do not make any successful changes, we recurse to the example's
         children and attempt the same there.
         """
-        self.example_wise_shrink(Length)
+        indices = [
+            (i, j)
+            for i, ls in enumerate(self.fine_grained_partitions)
+            for j in hrange(len(ls))
+        ]
+        self.random.shuffle(indices)
+
+        for i, j in indices:
+            if i >= len(self.fine_grained_partitions):
+                continue
+            partition = self.fine_grained_partitions[i]
+            # No point in trying to delete the last element because that will always
+            # give us a prefix.
+            if j >= len(partition) - 1:
+                continue
+
+            def clear_region(a, b):
```

Comment:
I'd consider renaming this function to `delete_region`, to make it more obvious that it will remove the indicated bytes rather than just zeroing them.